rfcs/0024-python-scope-package-set.md

```diff
@@ -0,0 +1,145 @@
+---
+feature: python-scope-package-set
+start-date: 2018-02-24
+author: Frederik Rietdijk
+co-authors:
+related-issues:
+---
+
+# Summary
+[summary]: #summary
+
+The Nix Packages collection (Nixpkgs) is a large collection of packages. A
+significant portion of these packages are written in the Python language or are
+to be used by a Python interpreter. A distinction is made in this RFC between
+*applications* that happen to be written in Python and *libraries* which are
+Python packages that are used in a development environment. This RFC formalizes
+and thereby clarifies this distinction and defines as well the scope of the
+*package set*, which is a curated set of compatible libraries that can be used
+in a development environment.
+
+# Motivation
+[motivation]: #motivation
+
+The motivation of this RFC is to formalize and thereby clarify the current way
+of working regarding the package set scope. The reason to have this formalized
+is to have a clear reference and guideline when making the daily decisions about
+what to include or exclude from the package set.
+
+# Detailed design
+[design]: #detailed-design
+
+## Introduction
+
+The Nix Packages collection (Nixpkgs) consists of a large amount of expressions
+of which a significant portion corresponds to package sets. Major package sets
+at the time of writing exist e.g. for the Haskell, Perl and Python languages.
+The scope and way working with these package sets depends on multiple factors,
+such as the tools provided upstream, the demand from the community and the
+maintainers of the package sets and the individual expressions. Another
+important factor is whether *upstream* provides a curated set of packages that
+are known to be compatible. An example of a curated set or snapshot is Stackage
+for Haskell. If such set exists and serves the needs from the community, then
+the scope may be limited to providing that set.
+
+## Definitions
+
+The following definitions are used in the scope of this RFC:
+- application: a package providing programs that happen to be written in Python
+- library: a package providing modules and programs that are used in a Python environment for development purposes or as dependency for applications
+- package set: a curated set of compatible libraries that can be used in a development environment and as dependencies for applications
+- maintainer: a maintainer of an expression as typically noted in the `meta.maintainers` field
+- package set maintainer: a maintainer of the Python package set as noted in the `CODEOWNERS` file
+
+Examples on the usage of application and library:
+- `scipy` is a collection of routines for scientific computing. It is used for development purposes or as a library and is therefore considered a *library*
+- `calibre` is a program for managing e-books that happens to be written in Python. The package is therefore considered an *application*
```

Comment:
On the other hand, https://manual.calibre-ebook.com/function_mode.html#the-power-of-function-mode-using-a-spelling-dictionary-to-fix-mis-hyphenated-words gives a snippet of code with
```
from calibre import replace_entities
from calibre import prepare_string_for_xml
```

Does this make Calibre _both_ application and library?

Reply:
Maybe pick an example that doesn't use `setuptools` then?